ast/dml.go

```diff
@@ -764,6 +764,8 @@ type SelectStmt struct {
 	IsAfterUnionDistinct bool
 	// IsInBraces indicates whether it's a stmt in brace.
 	IsInBraces bool
+	// QueryBlockOffset indicates the order of this SelectStmt if counted from left to right in the sql text.
+	QueryBlockOffset int
```

Comment:
Where would this field be set?

Reply:
In TiDB, see https://github.com/pingcap/tidb/pull/11861/files#diff-59f10b93675b42905c972e190d2f059dR45

Reply:
It looks pretty weird that the values is set from another repo, can we set it during the induction of parser?

Reply:
It seems that there is no way to set it correctly here, since the subpart of select stmt will get the offset first. I don't find good ways to set it correctly here.